src/averageAges.js

```diff
@@ -37,7 +45,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const isMother = name => (
+    people.some(person => person.mother === name)
+  );
+
+  const targetSelector = withChildren
+    ? person => (person.sex === 'f' && isMother(person.name))
+    : person => (person.sex === 'f');
+
+  const women = people.filter(targetSelector);
+  const womenAge = women.map(woman => woman.died - woman.born);
+
+  const womenAgeSum = womenAge.reduce((prev, current) => prev + current, 0);
```

Comment:
You can create a function like 'getSumOfAge', because you did twice the same map and reduce.